lld/ELF/InputFiles.cpp

```diff
@@ -741,6 +741,31 @@ template <class ELFT> void ObjFile<ELFT>::initializeJustSymbols() {
   sections.resize(numELFShdrs);
 }
 
+static bool isKnownSpecificSectionType(uint32_t t, uint32_t flags) {
+  if (SHT_LOUSER <= t && t <= SHT_HIUSER && !(flags & SHF_ALLOC))
+    return true;
+  if (SHT_LOOS <= t && t <= SHT_HIOS && !(flags & SHF_OS_NONCONFORMING))
+    return true;
+  switch (config->emachine) {
+  case EM_ARM:
+    return t == SHT_ARM_EXIDX || t == SHT_ARM_ATTRIBUTES;
+  case EM_AARCH64:
+    return t == SHT_AARCH64_MEMTAG_GLOBALS_STATIC;
```

Comment:
GNU ld will accept SHT_AARCH64_ATTRIBUTES in AArch64 too. This is reserved for the spec and can be ignored if a linker does not understand the contents. It would be useful to add this so that when  SHT_AARCH64_ATTRIBUTES is added lld doesn't choke on them.

This may be the case for other LOPROC/HIPROC values that lld doesn't currently understand, but I'm only familiar with Arm and AArch64.

Of the ones in AArch64 there is also `SHT_AARCH64_AUTH_RELR` defined in https://github.com/ARM-software/abi-aa/blob/main/pauthabielf64/pauthabielf64.rst#7section-types 

I know there is a pull-request to add LLD support for that. May be worth adding as a constant until the symbolic value is known.

The other section types in https://github.com/ARM-software/abi-aa/blob/main/aaelf32/aaelf32.rst not implemented by LLD don't need to be added as these are linker generated sections for the benefit of debuggers/post-link tools.

Reply:
Thanks. Sounds good to allow SHT_AARCH64_ATTRIBUTES as well even if it is unused now.
SHT_AARCH64_AUTH_RELR is for dynamic relocations only, so rejecting it here is fine. The test has a SHT_RELR example.

Shall we just allow all processor-specific relocation types?

Reply:
That's an interesting question. It is a pity there isn't a `SHF_PROC_NONCONFORMING` flag that would make this easier to reason about.

I think allowing processor section types would be an easier way to introduce this patch as it wouldn't need all the processor specific looked over, which may be difficult to do. Perhaps start with it this way, then as we get more information we could perhaps limit to the section types that we know exist in the specifications.

I know at Arm we are reluctant to introduce new section types because of the problem of older, and other proprietary linkers, that may not want to suppport them. I expect many other processor namespace ABIs think similarly.